projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-quick-order.ts

```
@@ -0,0 +1,92 @@
+
+  cy.visit(`/cart`);
+  // cy.wait(`@${cartPage}`).its('status').should('eq', 200);
+}
+
+export function visitQuickOrderPage() {
+  // const quickOrderPage = waitForPage(
+  //   '/my-account/quick-order',
+  //   'quickOrderPage'
+  // );
+
+  cy.visit('/my-account/quick-order');
+  // cy.wait(`@${quickOrderPage}`).its('status').should('eq', 200);
+}
+
+export function addProductToTheList(productCode: string) {
+  cy.get('.quick-order-form-input input').type(`${productCode}{enter}`);
+}
+
+export function addMultipleProductsToTheList(products: SampleProduct[]) {
+  products.forEach((product) => {
+    this.addProductToTheList(product.code);
+    // TODO
+    cy.wait(1500);
+  });
+}
+
+export function fillFormWithProductCode(product: SampleProduct) {
+  cy.get('.quick-order-form-input input').type(`${product.code}`);
+}
+
+export function clearForm() {
+  cy.get(`.quick-order-form-input cx-icon`).click();
+}
+
+export function clearList() {
+  cy.get(`.quick-order-footer .clear-button`).click();
+}
+
+export function removeFirstRow() {
+  cy.get(`cx-quick-order-list .quick-order-list-row`)
+    .first()
+    .find('.quick-order-list-item-action .cx-action-link')
+    .click();
+}
+
+export function addToCart() {
+  cy.get(`.quick-order-footer .add-button`).click();
+}
+
+export function verifyMiniCartQuantity(quantity: number) {
+  cy.get('cx-mini-cart .count').should('contain', quantity);
+}
+
+export function verifyQuickOrderListQuantity(quantity: number) {
+  cy.get('cx-quick-order-list')
+    .find('.quick-order-list-row')
+    .should('have.length', quantity);
+}
+
+export function addProductToCartWithQuickForm(productCode: string) {
+  cy.get('cx-cart-quick-form .input-product-code').type(`${productCode}`);
+  cy.get('cx-cart-quick-form .apply-coupon-button').click();
+}
+
+export function prepareCartWithProduct() {
+  this.visitQuickOrderPage();
+  this.addProductToTheList(sampleData.products[0].code);
+  this.addToCart();
+  cy.wait(1000);
```

instead of hardcoding the `wait` in which will most likely fail one day, you can 

* Cypress is very fast and one main reason that we have flaky tests is that we usually forget to wait for the response for a set of data that we're asserting to.

```
- create an intercept, and intercept the exact endpoint that will get triggered when you click add to cart
- wait for the response to give you a status of 200.

example: with saved cart, I intercepted the request that gets triggered when I click save a cart from the cart page

export function interceptSaveCartEndpoint(cartCode: string) {
  cy.intercept(
    'PATCH',
    `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
      'BASE_SITE'
    )}/users/current/carts/${cartCode}/save?*`
  ).as(SAVE_CART_ENDPOINT_ALIAS);

  return SAVE_CART_ENDPOINT_ALIAS;
}

----
how to use would be like 

1 - const variableName = interceptSaveCartEndpoint('someCartNumber')

2 - cy.get('save cart button').click();

3 - cy.wait(`@${alias}`).its('response.statusCode').should('eq', 200);

^The above helps make sure the we got the response back from what we requested.


```